packages/commonwealth/client/scripts/state/api/threads/editThread.ts

```diff
@@ -4,60 +4,89 @@ import MinimumProfile from 'models/MinimumProfile';
 
 interface EditThreadProps {
   address: string;
   chainId: string;
   threadId: number;
-  topicId: number;
-  kind: 'discussion' | 'link';
-  stage: string;
-  newBody: string;
-  newTitle: string;
+  // for edit profile
+  newBody?: string;
+  newTitle?: string;
   url?: string;
-  authorProfile: MinimumProfile;
+  authorProfile?: MinimumProfile;
+  // for editing thread locked status
+  readOnly?: boolean;
+  // for editing thread stage
+  stage?: string;
+  // for editing thread pinned status
+  pinned?: boolean;
+  // for editing thread spam status
+  spam?: boolean;
+  // for editing thread archive status
+  archived?: boolean;
+  // for editing thread topic
+  topicId?: number;
+  // for editing thread collaborators
+  collaborators?: {
+    toAdd?: number[];
+    toRemove?: number[];
+  };
 }
 
 const editThread = async ({
   address,
   chainId,
   threadId,
-  topicId,
-  kind,
-  stage,
+  // for edit profile
   newBody,
   newTitle,
   url,
   authorProfile,
+  // for editing thread locked status
+  readOnly,
+  // for editing thread stage
+  stage,
+  // for editing thread pinned status
+  pinned,
+  // for editing thread spam status
+  spam,
+  // for editing thread archived status
+  archived,
+  // for editing thread topic
+  topicId,
+  // for editing thread collaborators
+  collaborators,
 }: EditThreadProps) => {
-  const {
-    action = null,
-    session = null,
-    hash = null,
-  } = await app.sessions.signThread({
-    community: app.activeChainId(),
-    title: newTitle,
-    body: newBody,
-    link: url,
-    topic: topicId,
-  });
-
   const response = await axios.patch(`${app.serverUrl()}/threads/${threadId}`, {
+    // common payload
     author_chain: chainId,
-    author: JSON.stringify(authorProfile),
     address: address,
     chain: chainId,
-    kind: kind,
-    stage: stage,
-    body: encodeURIComponent(newBody),
-    title: encodeURIComponent(newTitle),
-    url,
     jwt: app.user.jwt,
-    canvas_action: action,
-    canvas_session: session,
-    canvas_hash: hash,
+    // for edit profile
+    ...(url && { url }),
+    ...(newBody && { body: encodeURIComponent(newBody) }),
+    ...(newTitle && { title: encodeURIComponent(newTitle) }),
+    ...(authorProfile && { author: JSON.stringify(authorProfile) }),
+    // for editing thread locked status
+    ...(readOnly !== undefined && { locked: readOnly }),
+    // for editing thread stage
+    ...(stage && { stage }),
+    // for editing thread pinned status
+    ...(pinned !== undefined && { pinned: pinned }),
+    // for editing thread spam status
+    ...(spam !== undefined && { spam: spam }),
+    // for editing thread archived status
+    ...(archived !== undefined && { archived: archived }),
+    // for editing thread topic
+    ...(topicId !== undefined && { topicId: topicId }),
+    // for editing thread collaborators
+    ...(collaborators !== undefined && { collaborators: collaborators }),
```

Comment:
everywhere you use { variable: variable } it can be just { variable }

Reply:
Nice spot. Fixed